### PR TITLE
feat: client overhaul proof of concept

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes nothing and almost certainly opens many things.

## What is this?

The beginnings of a migration from a static site built by Gatsby to a hybrid NextJS app.

## Why is this?

Static sites are fantastic for their specific use case, but are not very flexible.  

NextJS allows us to dynamically create pages, via [incremental static regeneration](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration), while still keeping most of the performance of a static site.

The fact that we no-longer have to build all pages ahead of time means that we can decouple content releases (i.e. changes to the curriculum) from client releases.  This decoupling allows for a more module architecture in which the curriculum can become a service. Rather than being simply data that is consumed during the build pipeline it can be a service that is queried on demand.  This means there's no longer a question of which application contains the definitive 'version' of the curriculum - the curriculum service will be definitive.

Finally, because NextJS makes dynamic routing a breeze.  We can create catch all routes like `/learn/[id]` to redirect users to the correct page immediately and easily.  All while serving as little to the user as possible.

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
It's not ready, obviously, but I'd love to get any feedback!

I don't have a concrete point at which I think this should be merged, but I do want to tidy it up a bit first.

**Review by ShaunSHamilton — COMMENTED** [review]
1) Is the `curriculum-server` a server? Or, is it just what is used during development?

If so, I would prefer it to go with the curriculum.

2) I am not a fan of calling something a `web` directory. Is this just common/standard Nextjs? Or, can we easily rename it to `client` (move it into `client)?

**Comment by ojeytonwilliams** [comment]
Nope, it's called web because it needed a name. Long term we can just rename it to client or whatever we want.

The curriculum server is, for now, just a dev thing. I haven't figured out what it's final form is going to be. It could be just a CDN or maybe we'll want something fancier (graphql? REST?).

It is being kept separate from the curriculum on purpose, though, simply because it's easier to keep them decoupled for the moment.

File: web/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+```sh
+npm ci
+npm run dev
+```
+
+Now the server should be running on port 3000 and the client on port 8000.
+
+For now there's not much to see.
+
+http://localhost:8000/responsive-web-design/special-path
```
Comment (moT01) [inline web/README.md:16]:
```suggestion
http://localhost:8000/learn/special-path
```

Comment (nhcarrigan) [reply web/README.md:16]: 🏎️

**Review by Sembauke — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Pull request merged**